replica/database.cc

```diff
@@ -2026,7 +2031,15 @@ future<> database::do_apply(schema_ptr s, const frozen_mutation& m, tracing::tra
 
     row_locker::lock_holder lock;
     if (!cf.views().empty()) {
-        lock = co_await cf.push_view_replica_updates(s, m, timeout, std::move(tr_state), get_reader_concurrency_semaphore());
+        auto lock_f = co_await coroutine::as_future(cf.push_view_replica_updates(s, m, timeout, std::move(tr_state), get_reader_concurrency_semaphore()));
+        if (lock_f.failed()) {
+            auto ex = lock_f.get_exception();
+            if (try_catch<timed_out_error>(ex)) {
```

Comment:
The `update_write_metrics` function uses `is_timeout_exception` which handles multiple types of timeout exceptions. To preserve the old behavior I think you should use it instead of `try_catch<timed_out_error>` (this applies to multiple places in the diff).

Reply:
done in v3